cmd/juju/machine/upgradeseries.go

```diff
@@ -70,6 +75,7 @@ type upgradeSeriesCommand struct {
 	modelcmd.IAASOnlyCommand
 
 	upgradeMachineSeriesClient UpgradeMachineSeriesAPI
+	leadershipClient           coreleadership.Pinner
```

Comment:
No issues but, if I am not mistaken, a similar pattern was rejected in the past in favor of adding additional methods to the existing API while moving common things to common. This may not be the exact same case but it seems close to me.

Reply:
I discussed this approach with @jameinel prior to the patch. I believe it is acceptable given that the same root API connection is used.